sdk/core/Azure.Core/src/RequestFailedException.cs

```diff
@@ -21,6 +21,7 @@ namespace Azure
     public class RequestFailedException : Exception, ISerializable
     {
         private const string DefaultMessage = "Service request failed.";
+        internal const string NoContentOnSuccessMessage = "Service request succeeded. Response content will not be included to avoid logging sensitive data.";
```

Comment:
nit: 

```suggestion
        internal const string NoContentOnSuccessMessage = "Service request succeeded. Response content is not included to avoid logging sensitive data.";
```

Reply:
I believe response content usually means the HTTP body - if we remove more than just the content from the exception message, we might want to change this to reflect that.